Enable google/cloud/storagetransfer on macOS

We cannot compile google/cloud/storagetransfer with the latest googleapis (current as-of 2022-04-25) because the protos define a `UID` enum, which generates a `UID_MAX` identifier, which is a macro on macOS:
```
In file included from bazel-out/darwin-fastbuild/bin/external/com_google_googleapis/google/storagetransfer/v1/transfer.pb.h:39:
bazel-out/darwin-fastbuild/bin/external/com_google_googleapis/google/storagetransfer/v1/transfer_types.pb.h:3485:24: error: expected member name or ';' after declaration specifiers
  static constexpr UID UID_MAX =
  ~~~~~~~~~~~~~~~~~~~~ ^
/Applications/Xcode_12.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/sys/syslimits.h:86:28: note: expanded from macro 'UID_MAX'
#define UID_MAX            2147483647U  /* max value for a uid_t (2^31-2) */
                           ^
```

Sigh.  The same thing happens for `GID_MAX`:
```
bazel-out/darwin-fastbuild/bin/external/com_google_googleapis/google/storagetransfer/v1/transfer_types.pb.h:3520:24: error: expected member name or ';' after declaration specifiers
  static constexpr GID GID_MAX =
  ~~~~~~~~~~~~~~~~~~~~ ^
/Applications/Xcode_12.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/sys/syslimits.h:78:28: note: expanded from macro 'GID_MAX'
#define GID_MAX            2147483647U  /* max value for a gid_t (2^31-2) */
                           ^
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -128,6 +128,11 @@ the APIs in these libraries are stable, and are ready for production use.
 
 - [Cloud Support API](/google/cloud/support/README.md)
 
+### [Storage Transfer Service](/google/cloud/storagetransfer/README.md)
+
+The library has been re-enabled on macOS. See
+[#8785](https://github.com/googleapis/google-cloud-cpp/issues/8785) for details.
+
 ## v2.10.0 - 2023-05
 
 ### New Libraries

--- a/ci/kokoro/windows/builds/quickstart-bazel.ps1
+++ b/ci/kokoro/windows/builds/quickstart-bazel.ps1
@@ -45,9 +45,8 @@ function Get-Released-Quickstarts {
         # The following quickstarts have problems building on Windows:
         #   TODO(#8145) - asset (TRUE/FALSE macros)
         #   TODO(#8125) - channel (DOMAIN macro)
-        #   TODO(#8785) - storagetransfer (UID_MAX/GID_MAX macros)
         #   TODO(#10737) - dialogflow_es triggers bug in Bazel 6.0.0
-        Where-Object { -not ("asset", "channel", "dialogflow_es", "storagetransfer" -contains $_) } |
+        Where-Object { -not ("asset", "channel", "dialogflow_es" -contains $_) } |
         # TODO(#9923) - compiling all quickstarts on Windows is too slow
         Get-Random -Count 10
     Pop-Location

--- a/ci/kokoro/windows/builds/quickstart-cmake.ps1
+++ b/ci/kokoro/windows/builds/quickstart-cmake.ps1
@@ -50,9 +50,7 @@ function Get-Vcpkg-Features {
             # TODO(#8145) - does not compile on Windows.
             "asset",
             # TODO(#8125) - does not compile on Windows.
-            "channel",
-            # TODO(#8785) - does not compile on Windows.
-            "storagetransfer" -contains $_) } |
+            "channel" -contains $_) } |
         # These are convenience features to refactor dependencies; they do not have quickstarts.
         Where-Object { -not ("googleapis", "grpc-common", "grafeas" -contains $_) }
 }

--- a/google/cloud/storagetransfer/BUILD.bazel
+++ b/google/cloud/storagetransfer/BUILD.bazel
@@ -42,11 +42,6 @@ cc_library(
     name = "google_cloud_cpp_storagetransfer",
     srcs = [":srcs"],
     hdrs = [":hdrs"],
-    # TODO(#8785): Enable on macOS.
-    target_compatible_with = select({
-        "@platforms//os:macos": ["@platforms//:incompatible"],
-        "//conditions:default": [],
-    }),
     visibility = ["//:__pkg__"],
     deps = [
         "//:common",

--- a/google/cloud/storagetransfer/CMakeLists.txt
+++ b/google/cloud/storagetransfer/CMakeLists.txt
@@ -14,16 +14,6 @@
 # limitations under the License.
 # ~~~
 
-# TODO(#8785): Enable on macOS.
-if (APPLE)
-    message(
-        WARNING
-            "Cannot build google/cloud/storagetransfer on Apple platforms."
-            " More details at https://github.com/googleapis/google-cloud-cpp/issues/8785 ."
-    )
-    return()
-endif ()
-
 include(GoogleapisConfig)
 set(DOXYGEN_PROJECT_NAME "Storage Transfer API C++ Client")
 set(DOXYGEN_PROJECT_BRIEF "A C++ Client Library for the Storage Transfer API")

